DeviceAdapters/DemoCamera/DemoCamera.cpp

```diff
@@ -3771,6 +3770,31 @@ void CDemoXYStage::ComputeIntermediatePosition(
    currentPosY = startPosY_um_ + fraction * (targetPosY_um_ - startPosY_um_);
 }
 
+void CDemoXYStage::CommitCurrentIntermediatePosition_(const MM::MMTime& now)
+{
+   if (timeOutTimer_ && !timeOutTimer_->expired(now))
+   {
+      // freeze where we *are* now
+      ComputeIntermediatePosition(now, posX_um_, posY_um_);
+   }
+   // No active motion → posX/Y already hold the last settled values
+
+   // Drop the timer so Busy() instantly goes idle
+   delete timeOutTimer_;
+   timeOutTimer_ = nullptr;
+
+   // Core listeners expect a position‑changed notification
+   (void)OnXYStagePositionChanged(posX_um_, posY_um_);
```

Comment:
Maybe better to avoid emitting this event if already stopped? Not that it is technically wrong.

Reply:
have another look